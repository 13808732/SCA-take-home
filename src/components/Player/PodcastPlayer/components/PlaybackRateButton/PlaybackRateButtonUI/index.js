import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import rgba from 'styles/helpers/rgba';
import screen from 'styles/helpers/media';

const StyledButton = styled.button`
  color: ${props => props.theme.light};
  background: transparent;
  transition: transform 0.2s ease, background-color 0.2s ease;
  border-style: solid;
  border-width: 1px;
  border-color:  ${props => rgba(props.theme.light, 0.3)};
  border-radius: 100%;
  padding: 0;
  cursor: ${props => (props.disable ? 'default' : 'pointer')};
  width: 40px;
  height: 40px;


  ${screen.md} {
    width: 112px;
    border-radius: 22px;
  }

  &:hover,
  &:active {
    outline: none;
  }

  &:focus {
    outline: none;
  }
`;

function PlaybackRateButtonUI({ rate, isPlaying, isLoading, onClick, variant, disable, ...props }) {
  return (
    <StyledButton onClick={onClick} title="Playback rate" disable={disable} {...props}>
      {`${rate}x`}
    </StyledButton>
  );
}

PlaybackRateButtonUI.propTypes = {
  rate: PropTypes.number,
  onClick: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool,
  isLoading: PropTypes.bool,
  disable: PropTypes.bool,
  variant: PropTypes.oneOf(['s', 'm', 'l']),
};

PlaybackRateButtonUI.defaultProps = {
  rate: 1,
  variant: 'm',
  isPlaying: false,
  isLoading: true,
  disable: false,
};

export default PlaybackRateButtonUI;
