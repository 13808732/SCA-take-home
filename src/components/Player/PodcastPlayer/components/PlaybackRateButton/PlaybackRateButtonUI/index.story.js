import React from 'react';
import { storiesOf } from '@storybook/react';
import PlayButtonUI from './index';

storiesOf('Play button', module)
  .addWithJSX('l, playing', () => (
    <PlayButtonUI
      className=""
      isPlaying
      isLoading={false}
      variant="m"
      onClick={() => {}}
    />
  ))
  .addWithJSX('m, paused', () => (
    <PlayButtonUI
      className=""
      isPlaying={false}
      isLoading={false}
      variant="l"
      onClick={() => {}}
    />
  ))
  .addWithJSX('s, loading', () => (
    <PlayButtonUI
      className=""
      isPlaying={false}
      isLoading
      variant="s"
      onClick={() => {}}
    />
  ));
