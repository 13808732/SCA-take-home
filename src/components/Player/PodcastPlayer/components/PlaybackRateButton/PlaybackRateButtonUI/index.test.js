import React from 'react';
import renderer from 'react-test-renderer';
import PlaybackRateButtonUI from './index';

describe('<PlaybackRateButtonUI /> component', () => {
  const props = {
    isPlaying: true,
    isLoading: false,
    onClick: jest.fn(),
  };

  test('it should render PlaybackRateButtonUI component', () => {
    const component = renderer.create(<PlaybackRateButtonUI {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
