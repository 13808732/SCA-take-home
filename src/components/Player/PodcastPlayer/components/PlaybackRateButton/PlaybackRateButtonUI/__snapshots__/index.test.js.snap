// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaybackRateButtonUI /> component it should render PlaybackRateButtonUI component 1`] = `
<button
  className="sc-bdnylx jyqamV"
  onClick={[MockFunction]}
  title="Playback rate"
>
  1x
</button>
`;
